### ISSUE TITLE
import extended streaming history error: "Expected string, received null"

### ISSUE THREAD

**Issue by jakesmorrison** [Yooooomi/your_spotify#291]
**Describe the bug**

When importing extended streaming history data I get the following error: 

```json
{
  "code": "invalid_type",
  "expected": "string",
  "received": "null",
  "path": [ "reason_end"],
  "message": "Expected string, received null"
}
```

```console
[error]  TypeError: Cannot read properties of null (reading 'length')
    at FullPrivacyImporter.<anonymous> (/app/lib/tools/importers/full_privacy.js:107:51)
    at Generator.next (<anonymous>)
    at fulfilled (/app/lib/tools/importers/full_privacy.js:5:58)
```

I requested my streaming history 3 weeks ago and received it today. The json naming is not "endsong". It is "Streaming_History_Audio". I have manually changed it to endsong, but I am getting a format error. I am guessing the output format has changed? 

**Expected behavior**

When I go from Settings -> Account -> Import Data, click "extended streaming history" and add my endsong.json I expect the data to be added to the your_spotify app. This process worked with 1 of my json files. 

**Additional context**

Docker compose

![image](https://github.com/Yooooomi/your_spotify/assets/14865171/968a47ab-a9ce-4024-9a5d-44bd280265a4)

**Shared link**

Let me figure out how to get a shared link. I will update with it.

**Comment by jakesmorrison** [comment]
I ran a python script to see what all the possible values of reason_end

```python
Reason End
['trackdone', 'playbtn', 'endplay', 'fwdbtn', 'backbtn', 'unexpected-exit-while-paused', 'clickrow', None, 'logout', 'unknown', 'unexpected-exit']
```
It looks like there is a None (Null). I ended up fixing this issue by cleaning up those values with this snippet. 
```python
import json
with open('endsong.json') as f:
    d = json.load(f)
    
for x in d:
    if x['reason_end'] == None:
        x['reason_end'] = 'endplay'
with open('endsong_clean.json', 'w') as f:
    json.dump(d, f)
```

**Comment by LucasGenoud** [comment]
> I ran a python script to see what all the possible values of reason_end
> 
> ```python
> Reason End
> ['trackdone', 'playbtn', 'endplay', 'fwdbtn', 'backbtn', 'unexpected-exit-while-paused', 'clickrow', None, 'logout', 'unknown', 'unexpected-exit']
> ```
> 
> It looks like there is a None (Null). I ended up fixing this issue by cleaning up those values with this snippet. After running the code I am able to import the data.
> 
> ```python
> import json
> with open('endsong.json') as f:
>     d = json.load(f)
>     
> for x in d:
>     if x['reason_end'] == None:
>         x['reason_end'] = 'endplay'
> with open('endsong_clean.json', 'w') as f:
>     json.dump(d, f)
> ```

Thank you ! 
Since i have multiple files and they don't necessarily end with endsong, I adapted your script slightly and I just update the original file:
`import json
import os

json_files = [file for file in os.listdir() if file.endswith('.json')]

for filename in json_files:
    with open(filename, 'r') as f:
        d = json.load(f)

    for x in d:
        if x['reason_end'] is None:
            x['reason_end'] = 'endplay'

    with open(filename, 'w') as f:
        json.dump(d, f)
`

**Comment by LucasGenoud** [comment]
Thank your ! 

I've adapted your script a little bit to read all the json files in the current folder since I have multiple file exports and they don't event end with endSong... I also just replace the original file

```
import json
import os

json_files = [file for file in os.listdir() if file.endswith('.json')]

for filename in json_files:
    with open(filename, 'r') as f:
        d = json.load(f)

    for x in d:
        if x['reason_end'] is None:
            x['reason_end'] = 'endplay'

    with open(filename, 'w') as f:
        json.dump(d, f)

```

**Comment by Yooooomi** [comment]
Should be fixed with latest release 1.7.1.
